code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -55,10 +55,14 @@
 		piping_layer = board.pipe_layer
 		set_layer = piping_layer
 
+	..() //Skipping the rest of on_construction() would be a bad idea so we clean up after it instead.
+
 	if(check_pipe_on_turf())
-		deconstruct(TRUE)
-		return
-	return..()
+		set_anchored(FALSE)
+		set_panel_open(TRUE)
+		change_pipe_connection(TRUE)
+		icon_state = "thermo-open"
+		visible_message(span_warning("The thermomachine is left unanchored as it's port is being hogged up by something else."))
```

Comment:
```suggestion
		balloon_alert(/* figure out the user here, visible_message should not be used as a hack */, "the port is already being used!")
```

Reply:
It's now a balloon alert...

Reply:
?? No it is not this, is the latest commit?

Reply:
Apparently I may have forgotten to sync it?